autorest/multiapi/__init__.py

```diff
@@ -0,0 +1,334 @@
+    - RT1 available on 2019-05-01 and 2019-06-01-preview
+    - RT2 available on 2019-06-01-preview
+    - RT3 available on 2019-07-01-preview
+
+    Then, if I put "RT2: 2019-06-01-preview" in the list, this means I have to make
+    "2019-06-01-preview" the default for models loading (otherwise "RT2: 2019-06-01-preview" won't work).
+    But this likely breaks RT1 default operations at "2019-05-01", with default models at "2019-06-01-preview"
+    since "models" are shared for the entire set of operations groups (I wished models would be split by operation groups, but meh, that's not the case)
+
+    So, until we have a smarter Autorest to deal with that, only preview RTs which do not share models with a stable RT can be added to this map.
+    In this case, RT2 is out, RT3 is in.
+    """
+
+    def there_is_a_rt_that_contains_api_version(rt_dict, api_version):
+        "Test in the given api_version is is one of those RT."
+        for rt_api_version in rt_dict.values():
+            if api_version in rt_api_version:
+                return True
+        return False
+
+    last_rt_list = {}
+
+    # Operation groups
+    versioned_dict = {
+        operation_name: [meta[0] for meta in operation_metadata]
+        for operation_name, operation_metadata in versioned_operations_dict.items()
+    }
+    # Operations at client level
+    versioned_dict.update(
+        {
+            operation_name: operation_metadata["available_apis"]
+            for operation_name, operation_metadata in mixin_operations.items()
+        }
+    )
+    for operation, api_versions_list in versioned_dict.items():
+        local_last_api_version = _get_floating_latest(api_versions_list, preview_mode)
+        if local_last_api_version == last_api_version:
+            continue
+        # If some others RT contains "local_last_api_version", and it's greater than the future default, danger, don't profile it
+        if (
+            there_is_a_rt_that_contains_api_version(
+                versioned_dict, local_last_api_version
+            )
+            and local_last_api_version > last_api_version
+        ):
+            continue
+        last_rt_list[operation] = local_last_api_version
+    return last_rt_list
+
+class MultiAPI:
+    def __init__(self, input_str: str, default_api: Optional[str] = None):
+        self.input_str = input_str
+        self.default_api = default_api
+
+    def process(self) -> bool:
+        # If True, means the auto-profile will consider preview versions.
+        # If not, if it exists a stable API version for a global or RT, will always be used
+        preview_mode = self.default_api and "preview" in self.default_api
+
+        # The only known multi-client package right now is azure-mgmt-resource
+        is_multi_client_package = "#" in self.input_str
+
+        package_name, module_name = _parse_input(self.input_str)
+        sdk_root = Path(__file__).parents[3] / "azure-sdk-for-python"
```

Comment:
Don't assume the SDK repo is cloned as siblings as autorest, please use 
`--python-sdks-folder=<path to the root directory of your azure-sdk-for-python clone>`

(add that in argparse)